### ISSUE TITLE
Rendering performance in Vector feels about 3-5x slower than the vanilla react example.

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#125]

**Comment by ara4n** [comment]
This is really painfully obvious when trying to resize the window (its trivial to max out 8 cores at 100% CPU just by rerendering), or using the app on Mobile Safari, where everything is sluggish as all hell.

**Comment by Half-Shot** [comment]
This is presumably no use to anyone anymore?

**Comment by turt2live** [comment]
It's slow.

**Comment by Half-Shot** [comment]
Do we not have a more targeted issue for "It's slow" than a issue about Vector in 2015 based off a old react example?

**Comment by turt2live** [comment]
tbh @lampholder would be the judge of how helpful an issue is, but in practice Riot is incredibly slow at rendering things (hundreds of milliseconds) which is problematic.

**Comment by Half-Shot** [comment]
Fair, I agree with the slow. I just think this isn't a great issue :)

**Comment by jryans** [comment]
I also agree this issue isn't great... It's not very actionable as it stands. 

It's quite general and non-specific, so even if many performance improvements were made, it would be hard to say if this issue could be closed. I think more targeted perf issues that focus on getting some area of code to meet some performance goal would be better way forward.

**Comment by ara4n** [comment]
https://github.com/vector-im/riot-web/issues/4997 is related, but i'd suggest keeping this open - it's literally "if you scrub the window around to resize it, it is incredibly sluggish to rerender"

**Comment by ara4n** [comment]
i suspect that killing the remaining gemini scrollbars with fire would help a lot.

**Comment by ara4n** [comment]
killing the remaining gemini scrollbars with fire did help a lot.

I still think we should profile the rendering properly though, relative to the original react example - my hunch is that there's something about our heavy use of nested flexbox that means that we're working CSS harder than we need to be, and strategically removing flexbox in a few places might speed things up enormously.

**Comment by gsouquet** [comment]
It seems that horizontal resizing of Element is pretty intense

![Screen Shot 2021-05-17 at 15 33 05](https://user-images.githubusercontent.com/769871/118506544-50ab2b00-b725-11eb-8100-e580784fff0f.png)

Resizing takes about 37ms with a pretty heavy layout operation being performed. This happens when being in an idle room. Anything else significant being dealt with at the same time is going to make the resizing experience even slower
